apps/system/lockscreen/js/lockscreen_inputpad.js

```diff
@@ -163,7 +167,19 @@
       var span = this.passcodeCode.childNodes[i];
       if (span) {
         if (this.states.passCodeEntered.length > i) {
+        var value = this.states.passCodeEntered.substr(i, this.states.passCodeEntered.length);
+        if ((this.states.passCodeEntered.length - 1 === i) && (keyCode && (keyCode !== 'b'))) {
+          (function(aspan, currentValue) {
+            aspan.textContent = currentValue;
+            setTimeout(function() {
+              aspan.textContent = '';
+              aspan.dataset.dot = true;
+            }, CODE_VISIBLE_TIMEOUT);
+          })(span, value);
+        } else {
+          span.textContent = '';
           span.dataset.dot = true;
+        }
```

Comment:
The indentation of the entire if clause above is wrong, as should be clear from the fact that there are two close braces at the same column on lines 182 and 183.